internal/service/ch/queries.go

```diff
@@ -0,0 +1,303 @@
+package ch
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/DIMO-Network/model-garage/pkg/vss"
+	"github.com/DIMO-Network/telemetry-api/internal/graph/model"
+	"github.com/volatiletech/sqlboiler/v4/drivers"
+	"github.com/volatiletech/sqlboiler/v4/queries"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+)
+
+const (
+	// IntervalGroup is the column alias for the interval group.
+	IntervalGroup = "group_timestamp"
+	tokenIDWhere  = vss.TokenIDCol + " = ?"
+	nameIn        = vss.NameCol + " IN ?"
+	timestampFrom = vss.TimestampCol + " > ?"
+	timestampTo   = vss.TimestampCol + " < ?"
+	sourceWhere   = vss.SourceCol + " = ?"
+	groupAsc      = IntervalGroup + " ASC"
+)
+
+// varibles for the last seen signal query.
+const (
+	lastSeenName = "'" + model.LastSeenField + "' AS name"
+	numValAsNull = "NULL AS " + vss.ValueNumberCol
+	strValAsNull = "NULL AS " + vss.ValueStringCol
+	lastSeenTS   = "max(" + vss.TimestampCol + ") AS ts"
+)
+
+// Aggregation functions for latest signals.
+const (
+	latestString    = "argMax(" + vss.ValueStringCol + ", " + vss.TimestampCol + ") as " + vss.ValueStringCol
+	latestNumber    = "argMax(" + vss.ValueNumberCol + ", " + vss.TimestampCol + ") as " + vss.ValueNumberCol
+	latestTimestamp = "max(" + vss.TimestampCol + ") as ts"
+)
+
+// Aggregation functions for float signals.
+const (
+	avgGroup  = "avg(" + vss.ValueNumberCol + ")"
+	randGroup = "groupArraySample(1, %d)(" + vss.ValueNumberCol + ")[1]"
```

Comment:
`randFloatGroup`, maybe?

We switch between number and float in a few places. Not a big deal.